### PR TITLE
add doc for new-collation-enable in br

### PR DESCRIPTION
**Pull request by joccau**
Signed-off-by: joccau <zak.zhao@pingcap.com>

### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)
Add comments for 6.0 br compatibility because of the default changing of the config  `new_collations_enabled_on_first_bootstrap` from `false` to `true`.

### Which TiDB version(s) do your changes apply to? (Required)
6.0 

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v6.0 (TiDB 6.0 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: br/backup-and-restore-tool.md [hunk 1]
```diff
@@ -97,6 +97,12 @@ BR 内置版本会在执行备份和恢复操作前，对 TiDB 集群版本和
 | 用 BR v4.0 备份 TiDB v4.0 | ✅ | ✅  | ✅（如果 TiKV >= v4.0.0-rc.1，BR 包含 [#233](https://github.com/pingcap/br/pull/233) Bug 修复，且 TiKV 不包含 [#7241](https://github.com/tikv/tikv/pull/7241) Bug 修复，那么 BR 会导致 TiKV 节点重启) |
 | 用 BR nightly 或 v5.0 备份 TiDB v4.0 | ❌（当 TiDB 版本小于 v4.0.9 时会出现 [#609](https://github.com/pingcap/br/issues/609) 问题) | ❌（当 TiDB 版本小于 v4.0.9 会出现 [#609](https://github.com/pingcap/br/issues/609) 问题) | ❌（当 TiDB 版本小于 v4.0.9 会出现 [#609](https://github.com/pingcap/br/issues/609) 问题) |
 
+#### new_collations_enabled_on_first_bootstrap 检查
```
Comment (3pointer) [inline br/backup-and-restore-tool.md:100]: 最好检查下 TiDB 相关文档是否有类似描述，可以做个关联

File: br/backup-and-restore-tool.md [hunk 2]
```diff
@@ -97,6 +97,12 @@ BR 内置版本会在执行备份和恢复操作前，对 TiDB 集群版本和
...
+
+从 TiDB v6.0 版本开始，`new_collations_enabled_on_first_bootstrap` 配置的默认值由 `false` 改为 `true`，当上下游集群的此项配置相同时，才可以将上游集群的备份数据安全地恢复到下游集群中。
+
+从 br v6.0 开始，br 会备份上游集群的 `new_collations_enabled_on_first_bootstrap` 配置项，同时在恢复时会检查此配置项是否与下游集群相同。如果用户需要将旧版本的备份数据恢复到 TiDB v6.0 中，用户需要自行检查上下游集群中的此配置项是否相同，若相同，则在恢复命令中添加 `--check-requirements=false` 可跳过此项配置检查。
```
Comment (TomShawn) [inline br/backup-and-restore-tool.md:104]: 如果上下游的 `new_collations_enabled_on_first_bootstrap` 配置项不相同会怎样呢？会有什么报错呢？

**Review by 3pointer — APPROVED** [review]
LGTM

File: br/backup-and-restore-tool.md (hunk 1 above)
Comment (TomShawn) [inline br/backup-and-restore-tool.md:100]:
```suggestion
#### 对 `new_collations_enabled_on_first_bootstrap` 进行检查
```

File: br/backup-and-restore-tool.md [hunk 3]
```diff
@@ -97,6 +97,12 @@ BR 内置版本会在执行备份和恢复操作前，对 TiDB 集群版本和
 | 用 BR v4.0 备份 TiDB v4.0 | ✅ | ✅  | ✅（如果 TiKV >= v4.0.0-rc.1，BR 包含 [#233](https://github.com/pingcap/br/pull/233) Bug 修复，且 TiKV 不包含 [#7241](https://github.com/tikv/tikv/pull/7241) Bug 修复，那么 BR 会导致 TiKV 节点重启) |
 | 用 BR nightly 或 v5.0 备份 TiDB v4.0 | ❌（当 TiDB 版本小于 v4.0.9 时会出现 [#609](https://github.com/pingcap/br/issues/609) 问题) | ❌（当 TiDB 版本小于 v4.0.9 会出现 [#609](https://github.com/pingcap/br/issues/609) 问题) | ❌（当 TiDB 版本小于 v4.0.9 会出现 [#609](https://github.com/pingcap/br/issues/609) 问题) |
 
+#### new_collations_enabled_on_first_bootstrap 检查
+
+从 TiDB v6.0 版本开始，[`new_collations_enabled_on_first_bootstrap`](/tidb-configuration-file.md#new_collations_enabled_on_first_bootstrap) 配置的默认值由 `false` 改为 `true`，当上下游集群的此项配置相同时，才可以将上游集群的备份数据安全地恢复到下游集群中，具体原因见[排序规则支持](/character-set-and-collation.md#排序规则支持)。
```
Comment (TomShawn) [inline br/backup-and-restore-tool.md:102]: [排序规则支持](https://github.com/joccau/docs-cn/blob/fix-check-collation/character-set-and-collation.md#%E6%8E%92%E5%BA%8F%E8%A7%84%E5%88%99%E6%94%AF%E6%8C%81) 里有提到 BR 不能在上下游该配置项不同时进行数据恢复的原因吗？
```suggestion
从 TiDB v6.0.0 版本开始，[`new_collations_enabled_on_first_bootstrap`](/tidb-configuration-file.md#new_collations_enabled_on_first_bootstrap) 配置项的默认值由 `false` 改为 `true`。当上下游集群的此项配置相同时，BR 才会将上游集群的备份数据安全地恢复到下游集群中。
```

File: br/backup-and-restore-tool.md [hunk 4]
```diff
@@ -97,6 +97,12 @@ BR 内置版本会在执行备份和恢复操作前，对 TiDB 集群版本和
...
+
+从 br v6.0 开始，br 会备份上游集群的 `new_collations_enabled_on_first_bootstrap` 配置项，同时在恢复时会检查此配置项是否与下游集群相同。若配置不相同，会拒绝恢复，并报告此配置项不匹配的错误。如果用户需要将旧版本的备份数据恢复到 TiDB v6.0 中，用户需要自行检查上下游集群中的此配置项是否相同，若相同，则在恢复命令中添加 `--check-requirements=false` 可跳过此项配置检查，若配置不相同的情况下强行恢复，会报告[数据校验错误](/br/backup-and-restore-tool.md#使用限制)。
```
Comment (TomShawn) [inline br/backup-and-restore-tool.md:104]:
```suggestion
从 v6.0.0 开始，BR 会备份上游集群的 `new_collations_enabled_on_first_bootstrap` 配置项，同时在恢复时会检查此配置项是否与下游集群相同。若上下游的该配置不相同，BR 会拒绝恢复，并报告此配置项不匹配的错误。

如果你需要将旧版本的备份数据恢复到 TiDB v6.0.0 或更新版本的 TiDB 集群中，你需要自行检查上下游集群中的该配置项是否相同：

- 若该配置该相同，则可在恢复命令中添加 `--check-requirements=false` 以跳过此项配置检查。
- 若该配置不相同，且进行强行恢复，BR 会报告[数据校验错误](/br/backup-and-restore-tool.md#使用限制)。
```

**Review by TomShawn — APPROVED** [review]
LGTM

**Pull request merged**